tests/cnf/core/network/sriov/tests/mlxSecureBoot.go

```diff
@@ -0,0 +1,195 @@
+			if !sriovenv.IsMellanoxDevice(sriovInterfacesUnderTest[0], workerNodeList[0].Object.Name) {
+				Skip("Mellanox Secure Boot test cases are supported only on Mellanox devices")
+			}
+
+			By("Enabling Mellanox firmware and wait for the cluster becomes stable")
+			err = sriovenv.ConfigureSriovMlnxFirmwareOnWorkersAndWaitMCP(
+				workerNodeList, sriovInterfacesUnderTest[0], true, totalVFs)
+			Expect(err).ToNot(HaveOccurred(), "Failed to configure Mellanox firmware")
+
+			By("Disabling Mellanox plugin in SriovOperatorConfig")
+			sriovOperatorConfig, err := sriov.PullOperatorConfig(APIClient, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to pull SR-IOV operator config")
+			_, err = sriovOperatorConfig.WithDisablePlugins([]string{"mellanox"}).Update()
+			Expect(err).ToNot(HaveOccurred(), "Failed to configure disablePlugins to include mellanox")
+
+			By("Enabling secure boot on the worker and reboot the node")
+			bmcClient = createBMCClient()
+
+			isSecureBootEnabled, err := bmcClient.IsSecureBootEnabled()
+			Expect(err).ToNot(HaveOccurred(), "Failed to validate if SecureBoot is enabled")
+
+			if !isSecureBootEnabled {
+				configureSecureBoot(bmcClient, "enable")
+			}
+		})
+
+		AfterAll(func() {
+			By("Removing SR-IOV configuration")
+			sriovOperatorConfig, err := sriov.PullOperatorConfig(APIClient, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to pull SR-IOV operator config")
+			_, err = sriovOperatorConfig.RemoveDisablePlugins().Update()
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete disablePlugins")
+
+			err = sriovenv.RemoveSriovConfigurationAndWaitForSriovAndMCPStable()
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove SR-IOV configuration")
+
+			By("Disabling secure boot on the worker and reboot the node")
+			isSecureBootEnabled, err := bmcClient.IsSecureBootEnabled()
+			Expect(err).ToNot(HaveOccurred(), "Failed to validate if SecureBoot is enabled")
+
+			if isSecureBootEnabled {
+				configureSecureBoot(bmcClient, "disable")
+			}
+
+			By("Disabling Mellanox firmware and wait for the cluster becomes stable")
+			err = sriovenv.ConfigureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, sriovInterfacesUnderTest[0], false, 0)
+			Expect(err).ToNot(HaveOccurred(), "Failed to configure Mellanox firmware")
+		})
+
+		It("End-to-End SR-IOV Configuration and Validation", reportxml.ID("77014"), func() {
+			By("Creating SR-IOV policy")
+			const sriovAndResourceNameSecureBoot = "securebootpolicy"
+
+			sriovPolicy := sriov.NewPolicyBuilder(
+				APIClient,
+				sriovAndResourceNameSecureBoot,
+				NetConfig.SriovOperatorNamespace,
+				sriovAndResourceNameSecureBoot,
+				2,
+				sriovInterfacesUnderTest[:1],
+				map[string]string{"kubernetes.io/hostname": workerNodeList[0].Definition.Name})
+
+			err := sriovenv.CreateSriovPolicyAndWaitUntilItsApplied(sriovPolicy, tsparams.MCOWaitTimeout)
```

Comment:
one more verification here. can you check that the node didn't reboot?
you can get the number of restarts on the config-daemon pod on the node and check it's equal after the MCP and nodeState are stable